core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/kubernetesModels.scala

```diff
@@ -0,0 +1,150 @@
+ */
+
+sealed abstract class KubernetesClusterStatus
+object KubernetesClusterStatus {
+  final case object StatusUnspecified extends KubernetesClusterStatus {
+    override def toString: String = "STATUS_UNSPECIFIED"
+  }
+
+  final case object Provisioning extends KubernetesClusterStatus {
+    override def toString: String = "PROVISIONING"
+  }
+
+  final case object Running extends KubernetesClusterStatus {
+    override def toString: String = "RUNNING"
+  }
+
+  final case object Reconciling extends KubernetesClusterStatus {
+    override def toString: String = "RECONCILING"
+  }
+
+  //in kubernetes statuses, they label "deleting' resources as 'stopping'
+  final case object Deleting extends KubernetesClusterStatus {
+    override def toString: String = "STOPPING"
+  }
+
+  final case object Error extends KubernetesClusterStatus {
+    override def toString: String = "ERROR"
+  }
+
+  final case object Degraded extends KubernetesClusterStatus {
+    override def toString: String = "DEGRADED"
+  }
+
+  //this is a custom status, and will not be returned from google
+  final case object Deleted extends KubernetesClusterStatus {
+    override def toString: String = "DELETED"
+  }
+
+  val deleted: KubernetesClusterStatus = Deleted
+  def values: Set[KubernetesClusterStatus] = sealerate.values[KubernetesClusterStatus]
+  def stringToObject: Map[String, KubernetesClusterStatus] = values.map(v => v.toString -> v).toMap
+}
+
+/** Google Container Nodepool statuses
+ * See https://googleapis.github.io/googleapis/java/all/latest/apidocs/com/google/container/v1/NodePool.Status.html
+ */
+sealed abstract class NodepoolStatus
+object NodepoolStatus {
+  case object StatusUnspecified extends NodepoolStatus {
+    override def toString: String = "STATUS_UNSPECIFIED"
+  }
+
+  case object Provisioning extends NodepoolStatus {
+    override def toString: String = "PROVISIONING"
+  }
+
+  case object Running extends NodepoolStatus {
+    override def toString: String = "RUNNING"
+  }
+
+  case object Reconciling extends NodepoolStatus {
+    override def toString: String = "RECONCILING"
+  }
+
+  //in kubernetes statuses, they label "deleting' resources as 'stopping'
+  case object Deleting extends NodepoolStatus {
+    override def toString: String = "STOPPING"
+  }
+
+  case object Error extends NodepoolStatus {
+    override def toString: String = "ERROR"
+  }
+
+  //this is a custom status, and will not be returned from google
+  case object Deleted extends NodepoolStatus {
+    override def toString: String = "DELETED"
+  }
+
+  case object RunningWithError extends NodepoolStatus {
+    override def toString: String = "RUNNING_WITH_ERROR"
+  }
+
+  def values: Set[NodepoolStatus] = sealerate.values[NodepoolStatus]
+  def stringToObject: Map[String, NodepoolStatus] = values.map(v => v.toString -> v).toMap
+}
+
+final case class KubernetesClusterLeoId(id: Long)
```

Comment:
`extend AnyVal` for all single param case classes so that they don't have extra boxing in runtime (in most cases)